cf_xarray/parametric.py

```diff
@@ -0,0 +1,815 @@
+
+    References
+    ----------
+    Please refer to the CF conventions document :
+      1. https://cfconventions.org/cf-conventions/cf-conventions.html#_ocean_sigma_over_z_coordinate
+    """
+
+    sigma: DataArray
+    eta: DataArray
+    depth: DataArray
+    depth_c: DataArray
+    nsigma: DataArray
+    zlev: DataArray
+
+    def decode(self) -> xr.DataArray:
+        """Decode coordinate.
+
+        Returns
+        -------
+        xr.DataArray
+            Decoded parametric vertical coordinate.
+        """
+        z_sigma = self.eta + self.sigma * (
+            np.minimum(self.depth_c, self.depth) + self.eta
+        )
+
+        z = xr.where(np.isnan(self.sigma), self.zlev, z_sigma)
+
+        return z.squeeze().assign_attrs(standard_name=self.computed_standard_name)
+
+    @property
+    def computed_standard_name(self) -> str:
+        """Computes coordinate standard name."""
+        return _derive_ocean_stdname(
+            eta=self.eta.attrs, depth=self.depth.attrs, zlev=self.zlev.attrs
+        )
+
+    @classmethod
+    def from_terms(cls, terms: dict):
+        """Create coordinate from terms."""
+        return cls(
+            **get_terms(terms, "sigma", "eta", "depth", "depth_c", "nsigma", "zlev")
+        )
+
+
+@dataclass
+class OceanDoubleSigma(ParamerticVerticalCoordinate):
+    """Ocean double sigma coordinate.
+
+    Standard name: ocean_double_sigma_coordinate
+
+    Parameters
+    ----------
+    sigma : xr.DataArray
+        Dimensionless coordinate.
+    depth : xr.DataArray
+        Distance (positive value) from datum to the sea floor.
+    z1 : xr.DataArray
+        Constant with units of length.
+    z2 : xr.DataArray
+        Constant with units of length.
+    a : xr.DataArray
+        Constant with units of length.
+    href : xr.DataArray
+        Constant with units of length.
+    k_c : xr.DataArray
+
+    References
+    ----------
+    Please refer to the CF conventions document :
+      1. https://cfconventions.org/cf-conventions/cf-conventions.html#_ocean_double_sigma_coordinate
+    """
+
+    sigma: DataArray
+    depth: DataArray
+    z1: DataArray
+    z2: DataArray
+    a: DataArray
+    href: DataArray
+    k_c: DataArray
+
+    def decode(self) -> xr.DataArray:
+        """Decode coordinate.
+
+        Returns
+        -------
+        xr.DataArray
+            Decoded parametric vertical coordinate.
+        """
+        f = 0.5 * (self.z1 + self.z2) + 0.5 * (self.z1 - self.z2) * np.tanh(
+            2 * self.a / (self.z1 - self.z2) * (self.depth - self.href)
+        )
+
+        z = xr.where(
+            self.sigma.k <= self.k_c,
+            self.sigma * f,
+            f + (self.sigma - 1) * (self.depth - f),
+        )
+
+        return z.squeeze().assign_attrs(standard_name=self.computed_standard_name)
```

Comment:
```suggestion
        return z.assign_attrs(standard_name=self.computed_standard_name)
```

Can we delete all the `squeeze` too please

Reply:
@dcherian Just want to check, if I remove the squeeze we'll end up with placeholder dimensions e.g. `(lev: 3, dim_0: 1, lat: 2, lon: 2)`. If that's fine then I'll go ahead and remove the call.

Reply:
where is the `dim_0` being created?

Reply:
@dcherian Circling back around, I've been on leave and just got a chance to look into this. It appears the `dim_0` was being generated by the dimensionless constants. It ends up `xr.DataArray([10], name='p0')` needed to be `xr.DataArray(10, name='p0')`. I've removed the squeeze calls and fixed the constant DataArrays.